src/System.ObjectModel/src/System/Collections/ObjectModel/KeyedCollection.cs

```diff
@@ -45,6 +47,19 @@ protected KeyedCollection(IEqualityComparer<TKey> comparer, int dictionaryCreati
             _threshold = dictionaryCreationThreshold;
         }
 
+        /// <summary>
+        /// Enables the use of foreach internally without allocations using <see cref="List{T}"/>'s struct enumerator.
+        /// </summary>
+        new private List<TItem> Items
+        {
+            get
+            {
+                Debug.Assert(base.Items.GetType() == typeof(List<TItem>));
```

Comment:
`is` is probably sufficient here, but I was just being very pedantic in the assert about `Items` being `List<T>` and not a subclass, which could have overridden the enumerator behavior.